src/flake8_numbers/check_numbers.py

```diff
@@ -0,0 +1,186 @@
+#! /usr/bin/env python3.11
+"""A flake8 plugin to check for numbers and their readability."""
+
+import ast
+from dataclasses import dataclass
+from typing import Any, Iterable, Optional, Tuple, Type
+
+
+@dataclass
+class ErrorReport:
+    """A class to represent an error report."""
+
+    line: int
+    """The line number of the error."""
+
+    column: int
+    """The column number of the error."""
+
+    message: str
+    """The error message."""
+
+
+class Flake8NumbersChecker:
+    """class to represent a flake8 plugin to check for numbers and their readability."""
+
+    name = "flake8-numbers"
+    version = "0.1"
+    off_by_default = False
+
+    def __init__(self, tree: ast.AST, filepath: str) -> None:
+        """Initialize the checker.
+
+        Args:
+            tree: The AST of the file being checked.
+            filepath: The path to the file being checked.
+        """
+        self._tree = tree
+        self._filepath = filepath
+
+    def run(self) -> Iterable[Tuple[int, int, str, Type[Any]]]:
+        """Run the checker.
+
+        Yields:
+            A tuple of the form (line, column, message, type).
+        """
+        for node in ast.walk(self._tree):
+            if isinstance(node, ast.Constant):
+                if isinstance(node.value, (int, float)):
+                    if result := self.check_constant(node):
+                        yield (
+                            result.line,
+                            result.column,
+                            result.message,
+                            Flake8NumbersChecker,
+                        )
+
+    def _extract_code(self, node: ast.AST) -> str:
+        """Extract the code of the given AST node.
+
+        Args:
+            node: The AST node to extract the code from.
+
+        Returns:
+            The code of the given AST node.
+        """
+        start_line, start_col = node.lineno, node.col_offset
+        end_line, end_col = node.end_lineno, node.end_col_offset
+        with open(self._filepath, "r", encoding="utf-8") as source_file:
+            lines = source_file.readlines()
+            original_code = "".join(
+                line[start_col:end_col].strip()
+                for line in lines[start_line - 1 : end_line]
+            )
+            return original_code
```

Comment:
It would be just `ast.Constant.value`, but then, lexic shougar like `_` are removed already. Hence, the much more complicated version, as the `ast` module is not giving me easy access to the "literal" value.